medusa/tv.py

```diff
@@ -2150,36 +2147,38 @@ def download_subtitles(self, lang=None):
 
         return new_subtitles
 
-    def check_for_meta_files(self, metadata_provider):
-        """Whether metadata files has changed.
+    def check_for_meta_files(self):
+        """Check Whether metadata files has changed. And write the current set self.hasnfo and set.hastbn.
 
-        :return:
+        :return: Whether a database update should be done on the episode.
         :rtype: bool
         """
         oldhasnfo = self.hasnfo
         oldhastbn = self.hastbn
 
-        cur_nfo = False
-        cur_tbn = False
+        all_nfos = []
+        all_tbns = []
 
         # check for nfo and tbn
-        if self.is_location_valid():
+        if not self.is_location_valid():
+            return False
+
+        for metadata_provider in app.metadata_provider_dict.values():
             if metadata_provider.episode_metadata:
                 new_result = metadata_provider.has_episode_metadata(self)
             else:
                 new_result = False
-            cur_nfo = new_result or cur_nfo
+            all_nfos += [new_result]
```

Comment:
I would use append() instead where possible, it's faster and more efficient.
all_nfos.append(new_result)